### PR TITLE
[three] Update to THREE.js r86 (0.86.0)

### PR DESCRIPTION
**Pull request by vcasagrande**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://threejs.org/docs/ https://github.com/mrdoob/three.js/tree/r86
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

#13418

### CODE REVIEW

File: types/three/three-core.d.ts [hunk 1]
```diff
@@ -6617,17 +6785,19 @@ export interface TextGeometryParameters {
...
 
     parameters: {
         font: Font;
-        size: number;
-        height: number;
-        curveSegments: number;
-        bevelEnabled: boolean;
-        bevelThickness: number;
-        bevelSize: number;
+        parameters: TextGeometryParameters;
```
Comment (sheetalkamat) [inline types/three/three-core.d.ts]: I see the paraemters of type { text: string; parameters: TextGeometryParameters } https://github.com/mrdoob/three.js/blob/ff2ce5657cf1e2567c7d240ff881b2564d133570/src/geometries/TextGeometry.js#L32

File: types/three/three-core.d.ts [hunk 2]
```diff
@@ -6617,17 +6785,19 @@ export interface TextGeometryParameters {
     bevelSize?: number;
 }
 
+export class TextBufferGeometry extends ExtrudeBufferGeometry {
+    constructor(text: string, parameters?: TextGeometryParameters);
+
+    parameters: TextGeometryParameters
```
Comment (sheetalkamat) [inline types/three/three-core.d.ts]: Dont see this.parameters assignment here? https://github.com/mrdoob/three.js/blob/ff2ce5657cf1e2567c7d240ff881b2564d133570/src/geometries/TextGeometry.js#L46

File: types/three/three-core.d.ts [hunk 3]
```diff
@@ -6846,6 +7016,20 @@ export class WireframeHelper extends LineSegments {
...
+
+    light: Light;
+    color: Color;
+
+    dispose(): void;
+    update(): void;
+}
+
+export class PolarGridHelper extends LineSegments {
+    constructor(radius: number, radials: number, circles: number, divisions: number, color1: number, color2: number);
```
Comment (sheetalkamat) [inline types/three/three-core.d.ts]: Arent all these parameters optional? https://github.com/mrdoob/three.js/blob/dcfa48a23524c4ecd5637cd9060ccc1413d2b429/src/helpers/PolarGridHelper.js#L16

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/three/three-core.d.ts [hunk 4]
```diff
@@ -2033,6 +2045,29 @@ export class SpotLight extends Light {
...
+ * This can be used to simulate things like bright windows or strip lighting.
+ */
+export class RectAreaLight extends Light {
+    /**
+     * @param color Hexadecimal color of the light. Default is 0xffffff (white)
+     * @param intensity Numeric value of the light's strength/intensity. Default is 1
+     * @param width Width of the light. Default is 10
+     * @param height Height of the light. Default is 10
+     */
+    constructor(color?: number|string, intensity?: number, width?: number, height?: number);
```
Comment (efokschaner) [inline types/three/three-core.d.ts]: looks like color can also be a `THREE.Color`,
It ends up here:
https://github.com/mrdoob/three.js/blob/74d0bdc64271f3e7f1b23c331a733433b1ea9915/src/math/Color.js#L37

File: types/three/three-core.d.ts [hunk 5]
```diff
@@ -2801,6 +2836,17 @@ export class MeshPhongMaterial extends Material {
     setValues(parameters: MeshPhongMaterialParameters): void;
 }
 
+/**
+ * An extension of the MeshPhongMaterial with toon shading
+ */
+export class MeshToonMaterial extends MeshPhongMaterial {
```
Comment (efokschaner) [inline types/three/three-core.d.ts]: There's also a `type` property that is undocumented but seems to be intentionally part of its public interface, see:
https://github.com/mrdoob/three.js/blob/e220a4baac6b415124cbfbf370f3c7aa44fbb13e/src/materials/MeshToonMaterial.js#L17

and for motivation:

https://threejs.org/docs/index.html#api/materials/Material

File: types/three/three-core.d.ts [hunk 6]
```diff
@@ -6440,17 +6551,37 @@ export class EdgesGeometry extends BufferGeometry {
...
+    curveSegments?: number;
+    steps?: number;
+    amount?: number;
+    bevelEnabled?: boolean;
+    bevelThickness?: number;
+    bevelSize?: number;
+    bevelSegments?: number;
+    extrudePath?: CurvePath<Vector>;
+    frames?: any;
+    UVGenerator?: any;
```
Comment (efokschaner) [inline types/three/three-core.d.ts]: It needs to match this: https://github.com/mrdoob/three.js/blob/61520f0d8a935f599e658119d7020e0525e17f35/src/geometries/ExtrudeGeometry.js#L756

File: types/three/three-core.d.ts [hunk 7]
```diff
@@ -6440,17 +6551,37 @@ export class EdgesGeometry extends BufferGeometry {
...
-    constructor(shape?: Shape, options?: any);
-    constructor(shapes?: Shape[], options?: any);
+    constructor(shape?: Shape, options?: ExtrudeGeometryParameters);
+    constructor(shapes?: Shape[], options?: ExtrudeGeometryParameters);
 
-    static WorldUVGenerator: {
-        generateTopUV(geometry: Geometry, indexA: number, indexB: number, indexC: number): Vector2[];
-        generateSideWallUV(geometry: Geometry, indexA: number, indexB: number, indexC: number, indexD: number): Vector2[];
-    };
+    addShapeList(shapes: Shape[], options?: ExtrudeGeometryParameters): void;
```
Comment (efokschaner) [inline types/three/three-core.d.ts]: agreed, don't think it should be deleted, and in fact you should extract its interface and share it with the `UVGenerator?: any` above.

**Comment by mhegazy** [comment]
Thanks for your contribution. This PR has unaddressed review commments and can not be merged in at the time being. For housekeeping purposes we are closing stale PRs. If you'd still like to merge this code in, please open a new PR that has been merged and rebased with the `master` branch.

**Pull request closed without merging**